src/awkward/_util.py

```diff
@@ -914,9 +914,7 @@ def apply(inputs, depth, user):
         elif any(isinstance(x, recordtypes) for x in inputs):
             if not allow_records:
                 raise ValueError(
-                    "cannot broadcast: {0}".format(
-                        ", ".join(repr(type(x)) for x in inputs)
-                    )
+                    "cannot broadcast records in this type of operation"
```

Comment:
This is not related to the fix, but when testing it, the original error message confused me.